addons/auth_signup/models/res_users.py

```diff
@@ -224,6 +239,71 @@ def send_unregistered_user_reminder(self, after_days=5):
             template = self.env.ref('auth_signup.mail_template_data_unregistered_users').with_context(dbname=self._cr.dbname, invited_users=invited_users[user])
             template.send_mail(user, email_layout_xmlid='mail.mail_notification_light', force_send=False)
 
+    def _should_alert_new_device(self):
```

Comment:
Would say that it deserves a small comment, to help people understand globally the condition in which we consider we should alert the user (like mfa_type being something else -> ?)